torch/csrc/jit/register_prim_ops.cpp

```diff
@@ -857,11 +858,11 @@ RegisterOperators reg(
                  "DictConstruct must have an even number of inputs");
            }
            return [=](Stack& stack) {
-             c10::ivalue::UnorderedMap vals;
+             c10::impl::GenericDict vals;
              for (size_t i = 0; i < num_inputs; i += 2) {
                auto val = pop(stack);
                auto key = pop(stack);
-               vals[key] = val;
+               vals.insert_or_assign(std::move(key), std::move(val));
```

Comment:
A map needs to supports indexing. This is really non-idiomatic both in C++ and Python. If anything Dict should have Python's API not C++'s since it used for custom ops that work in a subset of Python.

Reply:
This is unfortunately not possible given that the `Dict` implementation stores its keys and values as `IValues`. Let's discuss offline, maybe you have an idea.